src/plotly.nim

```diff
@@ -1,84 +1,110 @@
   ## create a plot with sane default layout.
   result = Plot[float64]()
-  result.datas = new_seq[Trace[float64]]()
-  result.layout = Layout(title: title, width: 600, height: 600, xaxis: Axis(title:xlabel), yaxis:Axis(title: ylabel), autosize:false)
+  result.datas = newSeq[Trace[float64]]()
+  result.layout = Layout(title: title, width: 600, height: 600,
+                         xaxis: Axis(title: xlabel),
+                         yaxis: Axis(title: ylabel),
+                         autosize: false)
 
-proc add*[T](p:Plot, d:Trace[T]) =
+proc add*[T](p: Plot, d: Trace[T]) =
   ## add a new data set to a plot.
   if p.datas == nil:
-    p.datas = new_seq[Trace[float64]]()
+    p.datas = newSeq[Trace[float64]]()
   p.datas.add(d)
 
-proc show*(p:Plot, path:string="", html_template:string=currentSourcePath().parentDir / "tmpl.html") =
-  var path = p.save(path, html_template)
+proc show*(p: Plot, path = "", html_template = defaultTmplPath) =
+  let path = p.save(path, html_template)
   browser.open(path)
   removeFile(path)
 
-proc save*(p:Plot, path:string="", html_template:string=currentSourcePath().parentDir / "tmpl.html"): string =
-  var ipath = path
-  if ipath == "":
-    ipath = "/tmp/x.html"
-  var jsons = new_seq[string]()
-  for d in p.datas:
-    jsons.add(d.json(as_pretty=true))
-  var data_string = "[" & join(jsons, ",") & "]"
-  var s = ($readFile(html_template)) % ["data", data_string, "layout", $(%p.layout), "title", p.layout.title]
-  var f:File
-  if not open(f, ipath, fmWrite):
+proc save*(p: Plot, path = "", html_template = defaultTmplPath): string =
+  result = path
+  if result == "":
+    result = "/tmp/x.html"
+  let
+    # call `json` for each element of `Plot.datas`
+    jsons = mapIt(p.datas, it.json(as_pretty = true))
+    data_string = "[" & join(jsons, ",") & "]"
+    # read the HTML template and insert data, layout and title strings
+    s = ($readFile(html_template)) % ["data", data_string, "layout", $(%p.layout),
+                                      "title", p.layout.title]
+  var
+    f: File
+  if not open(f, result, fmWrite):
     quit "could not open file for json"
   f.write(s)
   f.close()
-  return ipath
 
 when isMainModule:
   import math
 
-  var colors = @[Color(r:0.9, g:0.4, b:0.0, a: 1.0),
-                 Color(r:0.9, g:0.4, b:0.2, a: 1.0),
-                 Color(r:0.2, g:0.9, b:0.2, a: 1.0),
-                 Color(r:0.1, g:0.7, b:0.1, a: 1.0),
-                 Color(r:0.0, g:0.5, b:0.1, a: 1.0)]
-  var d = Trace[int](mode: PlotMode.LinesMarkers, `type`: PlotType.Scatter)
-  var size = @[16.int]
-  var m = Marker[int](size:size, color: colors)
+  let
+    # define a few colors
+    colors = @[Color(r:0.9, g:0.4, b:0.0, a: 1.0),
+               Color(r:0.9, g:0.4, b:0.2, a: 1.0),
+               Color(r:0.2, g:0.9, b:0.2, a: 1.0),
+               Color(r:0.1, g:0.7, b:0.1, a: 1.0),
+               Color(r:0.0, g:0.5, b:0.1, a: 1.0)]
+    # Trace object
```

Comment:
The formatting you introduce improves readability for sure, so I'll merge,
but adding e.g. comments like this convey no additional meaning. One can see it's a trace from the constructor. Same for e.g. colors above and markers below.

Reply:
haha, that's a good point, yes. sometimes a bad habit, because I want to explain something and end up with this...